tests/unit/pyrflx_test.py

```diff
@@ -272,33 +274,33 @@ def test_message_value_set_get_value(tlv_message_value: MessageValue) -> None:
 
 
 def test_message_value_get_invalid_field(tlv_message_value: MessageValue) -> None:
-    with pytest.raises(ValueError, match=r"field nofield not valid"):
+    with pytest.raises(PyRFLXError, match=r"pyrflx: error: field nofield not valid"):
         tlv_message_value.get("nofield")
 
 
 def test_message_value_set_invalid_field(tlv_message_value: MessageValue) -> None:
     tlv_message_value.set("Tag", "Msg_Data")
-    with pytest.raises(KeyError, match=r"cannot access field Value"):
+    with pytest.raises(PyRFLXError, match=r"pyrflx: error: cannot access field Value"):
         tlv_message_value.set("Value", b"")
-    with pytest.raises(KeyError, match=r"cannot access field Checksum"):
+    with pytest.raises(PyRFLXError, match=r"pyrflx: error: cannot access field Checksum"):
         tlv_message_value.set("Checksum", 8)
     tlv_message_value.set("Tag", "Msg_Error")
-    with pytest.raises(KeyError, match=r"cannot access field Length"):
+    with pytest.raises(PyRFLXError, match=r"pyrflx: error: cannot access field Length"):
         tlv_message_value.set("Length", 8)
 
 
 def test_message_value_invalid_value(tlv_message_value: MessageValue) -> None:
     with pytest.raises(
-        ValueError,
-        match="Error while setting value for field Tag: "
-        "cannot assign different types: str != int",
+        PyRFLXError,
+        match="pyrflx: error: cannot assign different types: str != int\n"
+        "pyrflx: error: cannot set value for field Tag",
     ):
         tlv_message_value.set("Tag", 1)
     tlv_message_value.set("Tag", "Msg_Data")
     with pytest.raises(
-        ValueError,
-        match="Error while setting value for field Length: "
-        "cannot assign different types: int != str",
+        PyRFLXError,
+        match="pyrflx: error: cannot assign different types: int != str\n"
+        "pyrflx: error: cannot set value for field Length",
```

Comment:
Wouldn't it make more sense to have the error (field cannot be set) before the explanation (type mismatch)? (That is a general question. There are more examples where the order changed.)

Reply:
I agree that would be better readable. 
In 5ce6adc the error messages are printed in reverse order (LIFO). Tests have been adapted in  d514a2e.

Reply:
I don't like that the string representation of `PyRFLXError` differs from its base class `RecordFluxError`. `PyRFLXError` is a `RecordFluxError` and should behave similarly. IMHO a changed error message order is rather unexpected. Is there reason for not fixing the order during creation of the affected `PyRFLXError`s?

Reply:
Changed it in 4f6d9d9 so that the most recent error entry can be added at the beginning of the list, which removes the need to print the entry list in reverse order.

Reply:
Maybe the `append_left` functionality could be moved directly into RecordFluxError (then `_error` could be a deque).

Reply:
Yes, I think moving the functionality into `RecordFluxError` and using a `deque` would be the better solution. I would also rename `append_left` to `appendleft` to make the interface more similar to Python's built-in datatypes (i.e. `deque`).

Reply:
Done in fe521d7